src/k8s-extension/azext_k8s_extension/partner_extensions/OpenServiceMesh.py

```diff
@@ -68,12 +65,12 @@ def Create(self, cmd, client, resource_group_name, cluster_name, name, cluster_t
             identity=None,
             location=""
         )
-        return extension_instance, name, create_identity
+        return extension, name, create_identity
 
-    def Update(self, extension, auto_upgrade_minor_version, release_train, version):
+    def Update(self, auto_upgrade_minor_version, release_train, version, configuration_settings,
+               configuration_protected_settings):
         """ExtensionType 'microsoft.openservicemesh' specific validations & defaults for Update
-           Must create and return a valid 'ExtensionInstanceUpdate' object.
-
+           Must create and return a valid 'PatchExtension' object.
         """
         #  auto-upgrade-minor-version MUST be set to False if release_train is staging or pilot
         if release_train.lower() in ['staging', 'pilot']:

```

Comment:
They removed this check from `Create` so I think it may be fine to remove this and use the `DefaultExtension` implementation for now